src/app/fsat/fan-field-data/fan-field-data.component.ts

```diff
@@ -162,21 +162,43 @@ export class FanFieldDataComponent implements OnInit {
   }
 
   calculateCompressibility() {
-    let tmpResults: FsatOutput = this.fsatService.getResults(this.fsat, true, this.settings);
-    //todo
+    let fsatOutput: FsatOutput;
+    let compressibilityFactor: number;
+    fsatOutput = this.fsatService.getResults(this.fsat, this.baseline, this.settings);
+
     let inputs: CompressibilityFactor = {
-      moverShaftPower: tmpResults.motorShaftPower,
+      moverShaftPower: fsatOutput.motorShaftPower,
       inletPressure: this.fieldDataForm.controls.inletPressure.value,
       outletPressure: this.fieldDataForm.controls.outletPressure.value,
       barometricPressure: this.fsat.baseGasDensity.barometricPressure,
       flowRate: this.fieldDataForm.controls.flowRate.value,
       specificHeatRatio: this.fieldDataForm.controls.specificHeatRatio.value
     };
-    let calcCompFactor: number = this.fsatService.compressibilityFactor(inputs, this.settings);
+
+    if (this.baseline) {
+      compressibilityFactor = this.fsatService.compressibilityFactor(inputs, this.settings);
+    } else {
+      let currentMoverShaftPower;
```

Comment:
Can you more this into a function that passes the parameters and returns the calculated value.